docs/go-configuration.md

```diff
@@ -4,27 +4,599 @@ title: Scope Go Agent configuration
+SCOPE_TRACER_GLOBAL=true
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithSetGlobalTracer()))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
 ```
 
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func main() {
+    scopeAgent, err := agent.NewAgent(agent.WithSetGlobalTracer())
+    if err != nil {
+        panic(err)
+    }
+    defer scopeAgent.Stop()
+
+    // ...
+}
+```
 
+<!--END_DOCUSAURUS_CODE_TABS-->
 
-## Change testing mode
+## Changing testing mode
 
 This variable indicates whether the agent is running tests or it is being used for runtime instrumentation. 
 
-If `SCOPE_TESTING_MODE` is not specified, the agent sets to  `true` when running in CI or if the tests are started using `scopeagent.Run()`,  and `false` otherwise.
+If `SCOPE_TESTING_MODE` is not specified, the agent sets to  `true` when running in CI or if the tests are started using `scopeagent.Run()` or if is instrumented using the `go.undefinedlabs.com/scopeagent/autoinstrument` import,  and `false` otherwise.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
 
 ```sh
 SCOPE_TESTING_MODE=false
 ```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithTestingModeEnabled()))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithTestingModeEnabled()))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Enabling test retries on failure
+
+The Scope agent has support for retrying failed tests. By default, no retries are performed.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_TESTING_FAIL_RETRIES=3
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithRetriesOnFail(3)))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithRetriesOnFail(3)))
```

Comment:
This doesn't seem like runtime instrumentation

Reply:
ok, my bad